Documentations/Doxegen/man/man3/BattleBoxFileManager.3
.TH "BattleBoxFileManager" 3 "Sat Jan 25 2020" "Battle Box Manual" \" -*- nroff -*-
.ad l
.nh
.SH NAME
BattleBoxFileManager
.SH SYNOPSIS
.br
.PP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBBattleBoxFileManager\fP (\fBBattleBoxFileManager\fP &&)=delete"
.br
.ti -1c
.RI "\fBBattleBoxFileManager\fP (const \fBBattleBoxFileManager\fP &)=delete"
.br
.ti -1c
.RI "\fBBattleBoxFileManager\fP & \fBoperator=\fP (\fBBattleBoxFileManager\fP &&)=delete"
.br
.ti -1c
.RI "\fBBattleBoxFileManager\fP & \fBoperator=\fP (const \fBBattleBoxFileManager\fP &)=delete"
.br
.in -1c
.SS "Static Public Member Functions"

.in +1c
.ti -1c
.RI "static void \fBDeleteFile\fP (const FString &sourcePath_)"
.br
.RI "Deletes a file\&. "
.ti -1c
.RI "static void \fBGetFileSize\fP (const FString &sourcePath_)"
.br
.RI "Gets the file size\&. "
.ti -1c
.RI "static bool \fBVerifyOnCreateDirectory\fP (const FString &testDir_)"
.br
.RI "Check and creates if there is a directory\&. "
.ti -1c
.RI "static void \fBWriteTextArrayToFile\fP (const FString &saveDirectory_, const TArray< FString > textArray_, const FString &FileName_, bool const IsOverWrite_)"
.br
.RI "write multiple lines of string to a txt file\&. "
.ti -1c
.RI "static void \fBWriteTextFile\fP (const FString &saveDirectory_, const FString context_, const FString &FileName_, bool const IsOverWrite_)"
.br
.RI "writes to a text file\&. "
.in -1c
.SH "Detailed Description"
.PP 
Definition at line 11 of file BattleBoxFileManager\&.h\&.
.SH "Member Function Documentation"
.PP 
.SS "void BattleBoxFileManager::DeleteFile (const FString & sourcePath_)\fC [static]\fP"

.PP
Deletes a file\&. 
.PP
Definition at line 35 of file BattleBoxFileManager\&.cpp\&.
.PP
.nf
36 {
37      if (!FPlatformFileManager::Get()\&.GetPlatformFile()\&.DeleteFile(*sourcePath_))
38      {
39           UE_LOG(LogTemp, Warning, TEXT("Could not find File!"));
40           return;
41      }
42 }
.fi
.SS "void BattleBoxFileManager::GetFileSize (const FString & sourcePath_)\fC [static]\fP"

.PP
Gets the file size\&. 
.PP
Definition at line 22 of file BattleBoxFileManager\&.cpp\&.
.PP
.nf
23 {
24      if(!FPlatformFileManager::Get()\&.GetPlatformFile()\&.FileSize(*sourcePath_)) 
25      {
26           UE_LOG(LogTemp, Warning, TEXT("File not Found!"));
27           return;
28      }
29      const int32 FileSize = FPlatformFileManager::Get()\&.GetPlatformFile()\&.FileSize(*sourcePath_);
30 
31      FString Message = FString::Printf(TEXT("File size is: %d"), FileSize);
32      UE_LOG(LogTemp, Log, TEXT("%s"), *Message);
33 
34 }
.fi
.SS "bool BattleBoxFileManager::VerifyOnCreateDirectory (const FString & testDir_)\fC [static]\fP"

.PP
Check and creates if there is a directory\&. 
.PP
Definition at line 8 of file BattleBoxFileManager\&.cpp\&.
.PP
.nf
9 {
10      IPlatformFile& PlatFromFile = FPlatformFileManager::Get()\&.GetPlatformFile();
11 
12      if (!PlatFromFile\&.DirectoryExists(*testDir_))
13      {
14           PlatFromFile\&.CreateDirectory(*testDir_);
15           if(!PlatFromFile\&.DirectoryExists(*testDir_))
16           {
17                return false;
18           }
19      }
20      return true;
21 }
.fi
.PP
Referenced by Debugger::InitDebug()\&.
.SS "void BattleBoxFileManager::WriteTextArrayToFile (const FString & saveDirectory_, const TArray< FString > textArray_, const FString & FileName_, bool const IsOverWrite_)\fC [static]\fP"

.PP
write multiple lines of string to a txt file\&. 
.PP
Definition at line 57 of file BattleBoxFileManager\&.cpp\&.
.PP
.nf
58 {
59      if (!IsOverWrite_)
60      {
61           if (!FPlatformFileManager::Get()\&.GetPlatformFile()\&.FileExists(*saveDirectory_))
62           {
63                UE_LOG(LogTemp, Warning, TEXT("This does not work"));
64                return;
65           }
66      }
67 
68      FString FinalString = "";
69      for (auto e : textArray_)
70      {
71           FinalString += e;
72           FinalString += LINE_TERMINATOR;
73      }
74      FString absoluteFilePath = saveDirectory_ + "/" + FileName_;
75      FFileHelper::SaveStringToFile(*FinalString, *absoluteFilePath);
76 }
.fi
.SS "void BattleBoxFileManager::WriteTextFile (const FString & saveDirectory_, const FString context_, const FString & FileName_, bool const IsOverWrite_)\fC [static]\fP"

.PP
writes to a text file\&. 
.PP
Definition at line 43 of file BattleBoxFileManager\&.cpp\&.
.PP
.nf
44 {
45      IPlatformFile& PlateFormFile = FPlatformFileManager::Get()\&.GetPlatformFile();
46 
47      if(PlateFormFile\&.CreateDirectory(*saveDirectory_))
48      {
49           FString absoluteFilePath = saveDirectory_ + "/" + FileName_;
50           
51           if (IsOverWrite_ || !PlateFormFile\&.FileExists(*absoluteFilePath))
52           {
53                FFileHelper::SaveStringToFile(*context_, *absoluteFilePath);
54           }
55      }
56 }
.fi


.SH "Author"
.PP 
Generated automatically by Doxygen for Battle Box Manual from the source code\&.


Plugins/Battle_Box/Source/Battle_Box/Private/BattleBoxFileManager.cpp
// Fill out your copyright notice in the Description page of Project Settings.

#include "BattleBoxFileManager.h"
#include "HAL/PlatformFilemanager.h"
#include "GenericPlatform/GenericPlatformFile.h"
#include "Misc/FileHelper.h"

bool BattleBoxFileManager::VerifyOnCreateDirectory(const FString& testDir_)
{
	IPlatformFile& PlatFromFile = FPlatformFileManager::Get().GetPlatformFile();

	if (!PlatFromFile.DirectoryExists(*testDir_))
	{
		PlatFromFile.CreateDirectory(*testDir_);
		if(!PlatFromFile.DirectoryExists(*testDir_))
		{
			return false;
		}
	}
	return true;
}
void  BattleBoxFileManager::GetFileSize(const FString& sourcePath_)
{
	if(!FPlatformFileManager::Get().GetPlatformFile().FileSize(*sourcePath_)) 
	{
		UE_LOG(LogTemp, Warning, TEXT("File not Found!"));
		return;
	}
	const int32 FileSize = FPlatformFileManager::Get().GetPlatformFile().FileSize(*sourcePath_);

	FString Message = FString::Printf(TEXT("File size is: %d"), FileSize);
	UE_LOG(LogTemp, Log, TEXT("%s"), *Message);

}
void  BattleBoxFileManager::DeleteFile(const FString& sourcePath_)
{
	if (!FPlatformFileManager::Get().GetPlatformFile().DeleteFile(*sourcePath_))
	{
		UE_LOG(LogTemp, Warning, TEXT("Could not find File!"));
		return;
	}
}
void  BattleBoxFileManager::WriteTextFile(const FString& saveDirectory_, const FString context_, const FString& FileName_, const int32 line_,  bool const IsOverWrite_)
{
	IPlatformFile& PlateFormFile = FPlatformFileManager::Get().GetPlatformFile();

	if(PlateFormFile.CreateDirectory(*saveDirectory_))
	{
		FString absoluteFilePath = saveDirectory_ + "/" + FileName_;
		if (IsOverWrite_|| !PlateFormFile.FileExists(*absoluteFilePath))
		{
			FFileHelper::SaveStringToFile(*context_ + line_, *absoluteFilePath);
		}
	}
}

